src/services/codefixes/convertToEs6Module.ts

```diff
@@ -5,10 +5,10 @@ namespace ts.codefix {
         getCodeActions(context) {
             const { sourceFile, program, preferences } = context;
             const changes = textChanges.ChangeTracker.with(context, changes => {
-                const moduleExportsChangedToDefault = convertFileToEs6Module(sourceFile, program.getTypeChecker(), changes, program.getCompilerOptions().target!, preferences);
+                const moduleExportsChangedToDefault = convertFileToEs6Module(sourceFile, program.getTypeChecker(), changes, program.getCompilerOptions().target!, getQuotePreference(sourceFile, preferences));
                 if (moduleExportsChangedToDefault) {
                     for (const importingFile of program.getSourceFiles()) {
-                        fixImportOfModuleExports(importingFile, sourceFile, changes, preferences);
+                        fixImportOfModuleExports(importingFile, sourceFile, changes, getQuotePreference(importingFile, preferences));

```

Comment:
what about the other preferences? why aren't we pass the other ones too?

Reply:
We're currently not using any other preferences; if we did we would need to create a `ProcessedPreferences` interface since we don't want to pass raw preferences, we want preferences with quote style inferred.

Reply:
so shouldn't it be :

```ts
{...preferences, quotePreference: getQuotePreference(prefrences)} 
```

Reply:
`quotePreference` in `UserPreferences` is a different type than `QuotePreference` -- and it should be, because we want the type checker to ensure that we've processed the preferences.

Reply:
but we are leaving behind all other preferences.. the next user preference we add will not be passed through.

Reply:
That's true, which means we would have to change the code to pass more parameters if that turned out to be necessary. Or I could create `ProcessedPreferences` interface right now if you prefer?